silverback/application.py

```diff
@@ -203,9 +257,11 @@ def on_(
         elif isinstance(container, ContractEvent) and isinstance(
             container.contract, ContractInstance
         ):
-            if self.get_event_handler(container.contract.address, container.abi.name):
+            input_filter = EventInputFilter.from_on_args(container, kwargs)
+
+            if self.get_event_handler(container.contract.address, container.abi.name, input_filter):
                 raise DuplicateHandlerError(
-                    f"event {container.contract.address}:{container.abi.name}"
+                    f"event {container.contract.address}:{container.abi.name}:{input_filter}"
```

Comment:
I think it might be nicer to do it more like `EventName(indexedArg1=some_value,indexedArg3=other_value)` where only the active filter kwargs are shown, otherwise would look different e.g. `Transfer(to=...)` or `Approved(owner=...)`, for everywhere we display this.
```suggestion
                    f"event {container.contract.address}:{container.abi.name}({input_filter})"
```

random thought: taskiq has the tagging system for tasks, wonder if we can leverage that for the filter kwargs

Reply:
some updates relating to labels in the newest release: https://github.com/taskiq-python/taskiq/releases/tag/0.11.0

Reply:
Updated the error message.

Not really sure what you're thinking about with labels.  Their docs are vague about labels.  Also, considering the time crunch it may be best to iterate on this later.